src/Http/Routing/src/Builder/EndpointRoutingApplicationBuilderExtensions.cs

```diff
@@ -42,10 +42,56 @@ public static IApplicationBuilder UseRouting(this IApplicationBuilder builder)
                 throw new ArgumentNullException(nameof(builder));
             }
 
+            return UseRouting(builder, true);
+        }
+
+        /// <summary>
+        /// Adds a <see cref="EndpointRoutingMiddleware"/> middleware to the specified <see cref="IApplicationBuilder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IApplicationBuilder"/> to add the middleware to.</param>
+        /// <param name="overrideEndpointRouteBuilder">Whether a new <see cref="EndpointRouteBuilder"/> should be created.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        /// <remarks>
+        /// <para>
+        /// A call to <see cref="UseRouting(IApplicationBuilder)"/> must be followed by a call to
+        /// <see cref="UseEndpoints(IApplicationBuilder, Action{IEndpointRouteBuilder})"/> for the same <see cref="IApplicationBuilder"/>
+        /// instance.
+        /// </para>
+        /// <para>
+        /// The <see cref="EndpointRoutingMiddleware"/> defines a point in the middleware pipeline where routing decisions are
+        /// made, and an <see cref="Endpoint"/> is associated with the <see cref="HttpContext"/>. The <see cref="EndpointMiddleware"/>
+        /// defines a point in the middleware pipeline where the current <see cref="Endpoint"/> is executed. Middleware between
+        /// the <see cref="EndpointRoutingMiddleware"/> and <see cref="EndpointMiddleware"/> may observe or change the
+        /// <see cref="Endpoint"/> associated with the <see cref="HttpContext"/>.
+        /// </para>
+        /// </remarks>
+        public static IApplicationBuilder UseRouting(this IApplicationBuilder builder, bool overrideEndpointRouteBuilder)
```

Comment:
overrideEndpointRouteBuilder..... is literal.

Reply:
Alternative ideas:
- `branchBuilder` or `branchRouteBuilder`
- `replaceBuilder` or `replaceRouteBuilder`
- `overrideBuilder` or `overrideRouteBuilder`
- ?

Reply:
I think if we agree that this approach makes sense, we can discuss the naming in the API review. I think I like the `overrideRouteBuilder` option the best though.

Reply:
The bool doesn't seem like a meaningful public parameter, anyone that wants the 'true' behavior will already be calling `UseRouting()`.

This should be a separate method name and the bool would be an internal API.
`UseRoutingAgain()` (🚲 🐑)

Reply:
Can we not make this an extension method and leave it as a plain old static method? I don't expect many people to want to call this overload.

Reply:
This was actually one of my earlier ideas! I wanted to call it app.ReRoute() or app.RerunRouting().

Reply:
I like the idea of a different name. ResumeRouting?